### PR TITLE
K-21 Backend Code Refactor

### PR DESCRIPTION
**Pull request by sduval215**
Backend Code Refactor

### CODE REVIEW

File: server/utils/queries.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+};
+
+/**
+ * QUERY FOR GENERAL PURPOSE SENATE COLLECTION QUERYING
+ * @param query=== ANY AVAILABLE QUERY NAME FROM DB (OPTIONAL or NULL)
+ * @param selections === SPECIFIC DATA TO BE PASSED BACK (OPTIONAL or NULL)
+ * @param cb == CALLBACK WITH RETURNED DATA
+ */
+utils.querySenate = (query, selections, cb) => {
+    Senate.find(query)
```
Comment (ravi-lattis) [inline server/utils/queries.js]: You see only model name varies for these two functions. Can you get model name also in the parameters and create a dynamic query?

File: server/config.js [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
 config.db.host = 'localhost';
 config.db.name = 'kapitol';
 config.db.house = 'house';
 config.db.senate = 'senate';
 
+//Environement variables
+config.env.TINMAN_PORT = process.env.KAPITOL_TINMAN_PORT;
+
+
 export default config;
```
Comment (ravirajsubramanian) [inline server/config.js]: Add a new line here.

File: server/utils/queries.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        .select(selections)
+        .exec((err, senateDocs) => {
+            return cb(senateDocs);
+        });
+}
+
+export default utils;
```
Comment (ravirajsubramanian) [inline server/utils/queries.js]: New line here.

File: server/utils/queries.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        })
+};
+
+/**
+ * QUERY FOR GENERAL PURPOSE SENATE COLLECTION QUERYING
+ * @param query=== ANY AVAILABLE QUERY NAME FROM DB (OPTIONAL or NULL)
+ * @param selections === SPECIFIC DATA TO BE PASSED BACK (OPTIONAL or NULL)
+ * @param cb == CALLBACK WITH RETURNED DATA
+ */
+utils.querySenate = (query, selections, cb) => {
```
Comment (ravirajsubramanian) [inline server/utils/queries.js]: You see that only model name is changing for these two methods. Why can't you get a model name in the parameters and write a single method for the find?

**Review by ravirajsubramanian — COMMENTED** [review]
@sduval215 I have left some comments. Can you pls look into it?

File: server/utils/queries.js (hunk 4 above)
Comment (sduval215) [reply server/utils/queries.js]: @ravirajthedeveloper good point. Went ahead and made those changes on the new commit.

**Review by ravirajsubramanian — APPROVED** [review]
Looks good. Shall I merge It?

**Pull request merged**